clients/client-java/src/main/java/org/apache/gravitino/client/GravitinoClient.java

```diff
@@ -389,6 +390,43 @@ public String[] listRoleNames() throws NoSuchMetalakeException {
     return getMetalake().listRoleNames();
   }
 
+  /**
+   * Grant privileges to a role.
+   *
+   * @param role The name of the role.
+   * @param privileges The privileges to grant.
+   * @param object The object is associated with privileges to grant.
+   * @return The role after granted.
+   * @throws NoSuchRoleException If the role with the given name does not exist.
+   * @throws NoSuchMetadataObjectException If the metadata object with the given name does not
+   *     exist.
+   * @throws NoSuchMetalakeException If the Metalake with the given name does not exist.
+   * @throws RuntimeException If revoking roles from a group encounters storage issues.
+   */
+  public Role grantPrivilegeToRole(String role, MetadataObject object, List<Privilege> privileges)
+      throws NoSuchRoleException, NoSuchMetadataObjectException, NoSuchMetalakeException {
```

Comment:
I think we should also add an exception about failed to grant/revoke privileges to role, because some privileges cannot be bound to some metadata objects.

Reply:
I used IllegalArgumentException. WDYT?

Reply:
OK.

Reply:
On a second thought, maybe we should explicitly define an exception for illegal grant/revoke action, so that users will know better what issue it is, not implicitly thrown an exception here. Even for `IllegalArgumentException` I would suggest clearly define it.

Reply:
OK, I will do it.